livekit-agents/livekit/agents/beta/workflows/dtmf_inputs.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+import logging
+
+from typing import Callable, Optional
+
+from livekit import rtc
+from livekit.agents.voice.events import AgentStateChangedEvent, UserStateChangedEvent
+
+from ... import function_tool
+from ...job import get_job_context
+from ...llm.chat_context import ChatContext
+from ...types import NOT_GIVEN, NotGivenOr
+from ...utils.aio.debounce import Debounced, debounced
+from ...voice.agent import AgentTask
+from ..tools.dtmf import DtmfEvent, format_dtmf
+
+logger = logging.getLogger("dtmf-inputs")
+
+
+class GetDtmfTask(AgentTask[Optional[str]]):
+    """A task to collect DTMF inputs from the user.
+
+    Return a string of DTMF inputs if collected successfully, otherwise None.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        num_digits: int,
+        *,
+        chat_ctx: NotGivenOr[ChatContext] = NOT_GIVEN,
+        input_timeout: float = 5.0,
+        ask_for_confirmation: bool = False,
+        interrupt_on_complete_input: bool = False,
+    ) -> None:
```

Comment:
```suggestion
    def __init__(
        self,
        *,
        num_digits: int,
        extra_instructions: str.
        chat_ctx: NotGivenOr[ChatContext] = NOT_GIVEN,
        ask_for_confirmation: bool = False,
        dtmf_input_timeout: float = 5.0,
    ) -> None:
```
Let's always interrupt, no need for `interrupt_on_complete_input` IMO

We can replace `name` with `extra_instructions`, it's more flexible